great_expectations/expectations/metrics/metric_provider.py

```diff
@@ -96,14 +96,18 @@ class MetricProvider(metaclass=MetaMetricProvider):
 
     domain_keys: Tuple[str, ...] = tuple()
     value_keys: Tuple[str, ...] = tuple()
-    default_kwarg_values = {}
+    default_kwarg_values: dict = {}
 
     @classmethod
     def _register_metric_functions(cls) -> None:
         metric_name = getattr(cls, "metric_name", None)
         metric_domain_keys = cls.domain_keys
         metric_value_keys = cls.value_keys
 
+        if metric_name is None:
+            # No metric name has been defined
+            return
```

Comment:
```suggestion
        if not metric_name:
            # No metric name has been defined
            return
```

I expect we want an empty string to hit this branch as well?

Reply:
@Kilo59 Good idea (applying in next PR to be updated in a few minutes).  Thanks.